src/test/java/com/loofah/graph/api/retrievers/MongoBackedGradeRepositoryTest.java

```diff
@@ -0,0 +1,140 @@
+package com.loofah.graph.api.retrievers;
+
+import com.loofah.graph.api.models.database.Category;
+import com.loofah.graph.api.models.database.Craft;
+import com.loofah.graph.api.models.database.Grade;
+import com.loofah.graph.api.models.database.Skill;
+import com.loofah.graph.api.models.filters.SkillFilter;
+import com.loofah.graph.api.providers.MongoQueryProvider;
+import com.loofah.graph.api.repositories.CategoryRepository;
+import com.loofah.graph.api.repositories.CraftRepository;
+import com.loofah.graph.api.repositories.GradeRepository;
+import com.loofah.graph.api.repositories.SkillRepository;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+import org.springframework.data.mongodb.core.query.Query;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class MongoBackedGradeRepositoryTest {
+
+    @Mock
+    private MongoQueryProvider mongoQueryProvider;
+
+    @Mock
+    private SkillRepository skillRepository;
+
+    @Mock
+    private CategoryRepository categoryRepository;
+
+    @Mock
+    private GradeRepository gradeRepository;
+
+    @Mock
+    private CraftRepository craftRepository;
+
+    @InjectMocks
+    private MongoBackedDataRetriever mongoBackedDataRetriever;
+
+    @Test
+    public void getSkillById_getsSkillViaDataRetriever() {
```

Comment:
Nitpick: should these be `_getsXViaRepository`? (or I guess `getsSkillsViaMongoQuery` or something for the test below this one)

Reply:
yep - updated :)